### ISSUE TITLE
sphinx-quickstart prompts overflow on Console with 80 chars width

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3364]
Subject: when I run sphinx-quickstart in a Terminal window of 80 character widths, some prompts become barely readable, as seen in this screenshot.

![capture d ecran 2017-01-21 a 18 04 18](https://cloud.githubusercontent.com/assets/2589111/22176174/5ac01d1e-e004-11e6-969f-78bbdc2944d1.png)

Notice how `intersphinx` in the third prompt is mangled. Also the prompt cursor appears in disturbing location.

With a wider Terminal there is  no issue:

![capture d ecran 2017-01-21 a 18 02 16](https://cloud.githubusercontent.com/assets/2589111/22176175/66fd1b86-e004-11e6-9289-9877483d0589.png)

<strike>In the first case, enlarging the Terminal is too late.</strike> Actually enlarging the Terminal window during the prompting can cure the problem from user point of view.

#### Procedure to reproduce the problem
```
$ sphinx-quickstart
```

If I go to file `sphinx/quickstart.py` and make there the prompts shorter the problem disappears.

This is naturally a minor inconvenience: I can live with a wider Terminal !

But I would like to know if `do_prompt` function in `sphinx/quickstart.py` is optimal.

### Environment info
- OS: Mac os x 10.9.5
- Python version: 2.7.13 (or 3.5.2)
- Sphinx version: 1.5.1
- Terminal application in mac os x.

**Comment by tk0miya** [comment]
In my short investigation, it might be a bug of `raw_input()` function of Python.

I'd tested following code in narrow window:
```
>>> # with escape sequence
... raw_input('\x1b[35m> doctest: automatically test code snippets in doctest blocks (y/n) [n]: \x1b[39;49;00m')
blocks (y/n) [n]: ically test code snippets in doctest b
''
>>> # => broken
```
```
>>> # without escape sequence
... raw_input('> doctest: automatically test code snippets in doctest blocks (y/n) [n]: ')
> doctest: automatically test code snippets in doctest blocks (y/n) [n]:
''
>>> # => OK
```

As a ad-hoc fix, we should use other way to display prompt like following:
```
>>> sys.stderr.write('\x1b[35m> doctest: automatically test code snippets in doctest blocks (y/n) [n]: \x1b[39;49;00m'); raw_input('')
> doctest: automatically test code snippets in doctest blocks (y/n) [n]:
''
```

@shimizukawa Do you know any good solution?

**Comment by shimizukawa** [comment]
IMO, print(msg) + input() without msg is good solution for a while.

**Comment by tk0miya** [comment]
Fixed at 678cc3d :-)
Thanks!
